internal/provider/data_source_packer_image.go

```diff
@@ -33,16 +35,23 @@ func dataSourcePackerImage() *schema.Resource {
 				Type:        schema.TypeString,
 				Required:    true,
 			},
-			"iteration_id": {
-				Description: "HCP ID of this image.",
-				Type:        schema.TypeString,
-				Required:    true,
-			},
 			"region": {
 				Description: "Region this image is stored in, if any.",
 				Type:        schema.TypeString,
 				Required:    true,
 			},
+			// Optional inputs
+			"iteration_id": {
+				Description:  "HCP ID of this image. Either this or `channel' must be specified.",
+				Type:         schema.TypeString,
+				Optional:     true,
```

Comment:
What do you think about making this value `Compute` and `Optional`and setting this as an output of this data source as well? This way, users are able to know what's the iteration being queried

Reply:
This actually already set but in the output from the build response, in this case, it makes sense to make this one compute.

Reply:
I agree this one can be promoted to both `Optional` and `Compute` since it's either copied from the input, or derived from the channel, I'll do this if the Schema lets me